buildSrc/src/main/kotlin/org/cqfn/save/generation/Generation.kt

```diff
@@ -0,0 +1,196 @@
+    lateinit var description: String
+    lateinit var default: String
+}
+
+fun main() {
+    val configFile = "buildSrc/src/main/kotlin/config-options.json"
+    val gson = Gson()
+    val bufferedReader: BufferedReader = File(configFile).bufferedReader()
+    val jsonString = bufferedReader.use { it.readText() }
+    val jsonObject = gson.fromJson<Map<String, Option>>(jsonString, object : TypeToken<Map<String, Option>>(){}.type)
+    generateSaveProperties(jsonObject)
+    generateReadme(jsonObject)
+}
+
+fun generateSaveProperties(jsonObject: Map<String, Option>) {
+    val builder = FileSpec.builder("org.cqfn.save.core.config", "SaveProperties")
+    builder.addComment(autoGenerationComment)
+    builder.addImport("kotlinx.cli", "ArgParser")
+    builder.addImport("kotlinx.cli", "ArgType")
+    val classBuilder = generateSavePropertiesClass(jsonObject)
+    val mergeFunc = generateMergeConfigFunc(jsonObject)
+    classBuilder.addFunction(mergeFunc.build())
+    builder.addType(classBuilder.build())
+    builder.build().writeTo(System.out)
+    File("$generatedSaveSavePropertiesPath/SaveProperties.kt").writeText(builder.build().toString())
+}
+
+fun generateSavePropertiesClass(jsonObject: Map<String, Option>): TypeSpec.Builder {
+    val classBuilder = TypeSpec.classBuilder("SaveProperties")
+    var properties = ""
+    jsonObject.forEach { properties += ("@property ${it.key} ${it.value.description}\n")}
+    val kdoc = """
+               |Configuration properties of save application, retrieved either from properties file
+               |or from CLI args.
+               |$properties
+               """.trimMargin()
+    classBuilder.addKdoc(kdoc)
+    classBuilder.addAnnotation(AnnotationSpec.builder(ClassName("kotlinx.serialization", "Serializable")).build())
+    // Generate primary ctor
+    val primaryCtor = FunSpec.constructorBuilder()
+    for ((name, value) in jsonObject) {
+        primaryCtor.addParameter(ParameterSpec.builder(name, createClassName(value.kotlinType).copy(nullable = true))
+            .defaultValue(run {
+                var default = value.default
+                if (default != "null" && value.kotlinType == "kotlin.String") {
+                    default = "\"$default\""
+                }
+                default
+            })
+            .build())
+        val property = PropertySpec.builder(name, createClassName(value.kotlinType).copy(nullable = true))
+            .initializer(name)
+            .mutable()
+        classBuilder.addProperty(property.build())
+    }
+    classBuilder.primaryConstructor(primaryCtor.build())
+    // Generate secondary ctor
+    val secondaryCtor = FunSpec.constructorBuilder()
+    secondaryCtor.addParameter("args", ClassName("kotlin", "Array")
+                                                .parameterizedBy(ClassName("kotlin", "String")))
+    secondaryCtor.callThisConstructor()
+    secondaryCtor.addStatement("    val parser = ArgParser(\"save\")")
```

Comment:
Do you need to specify indent here? I thought kotlinpoet does this automatically

Reply:
Add option .indent() for whole class. But some statements now get specific indents from kotlinpoet